runtime/src/commonMain/kotlin/app/cash/sqldelight/Transacter.kt

```diff
@@ -69,6 +94,16 @@ interface Transacter {
     body: TransactionWithoutReturn.() -> Unit
   )
 
+  suspend fun <R> suspendingTransactionWithResult(
```

Comment:
My only concern with this is exposing this API when folks have not turned async mode in SQLDelight on.

A potential approach could be a subtype of `Transacter` and `TransacterImpl`, `SuspendingTransacter` and `SuspendingTransacterImpl` (could still be in this module, thats fine), which the generated types extend instead of the current types. Thoughts?